Gems/ROS2/Code/Source/RobotImporter/Utils/RobotImporterUtils.cpp

```diff
@@ -549,4 +613,42 @@ namespace ROS2::Utils::SDFormat
     {
         return supportedPlugins.contains(GetPluginFilename(plugin));
     }
+
+    sdf::ParserConfig CreateSdfParserConfigFromSettings(const SdfAssetBuilderSettings& settings)
+    {
+        sdf::ParserConfig sdfConfig;
+
+        sdfConfig.URDFSetPreserveFixedJoint(settings.m_urdfPreserveFixedJoints);
+
+        // Fill in the URI resolution with the supplied prefix mappings.
+        for (auto& [prefix, pathList] : settings.m_resolverSettings.m_uriPrefixMap)
+        {
+            std::string uriPath;
+            for(auto& path : pathList)
+            {
+                if (!uriPath.empty())
+                {
+                    uriPath.append(std::string(":"));
+                }
+
+                uriPath.append(std::string(path.c_str(), path.Native().size()));
+            }
+            if (!prefix.empty() && !uriPath.empty())
+            {
+                std::string uriPrefix(prefix.c_str(), prefix.size());
+                sdfConfig.AddURIPath(uriPrefix, uriPath);
+                AZ_Info("SdfParserConfig", "Added URI mapping '%s' -> '%s'", uriPrefix.c_str(), uriPath.c_str());
```

Comment:
Well based on the SDFormat documentation for AddURIPath, I guess this will never work for Windows style absolute paths.

I am guessing Robotic model import usage is pretty much doesn't occur on Windows?

Reply:
Yeah, I would guess that as well. There's very little about the ROS / Gazebo stuff so far that seems very tolerant of running under Windows. I'm sure it's _possible_ but painful and unusual enough that it seems pretty unlikely to be done very much in practice.